modules/templates/proxy_request.rb

```diff
@@ -0,0 +1,55 @@
+require "proxy/util"
+require 'proxy/request'
+require 'uri'
+
+module Proxy::Templates
+  class ProxyRequest < ::Proxy::HttpRequest::ForemanRequest
+    include Proxy::Log
+
+    def call_template(method, path, env, params, body = '')
+      template_url = Proxy::Templates::Plugin.settings.template_url
+      proxy_ip = URI.parse(template_url).host
+      opts = params.clone.merge(:url => template_url)
+      opts.delete("path")
+      opts.delete("template")
+      opts.delete("kind")
+      opts.delete("hostgroup")
+      opts.delete("splat")
+      opts.delete("captures")
+      # in hostgroup provisioning there are spaces
+      path = path.map { |x| CGI.escape(x) }.join('/')
+      logger.debug "Template: request for #{path} using #{opts.inspect} at #{uri.host}"
+      proxy_headers = extract_request_headers(env)
+      proxy_headers["X-Forwarded-For"] = "#{env['REMOTE_ADDR']}, #{proxy_ip}"
+      proxy_headers["Content-Type"] = params["Content-Type"] if params["Content-Type"]
```

Comment:
Perhaps there are more headers to pass along? `Accept` comes to mind`. Also note that HTTP headers are case insensitive. I don't know if `params` is case insensitive.

Reply:
I would not mind not doing this in this patch, but up to you @timogoebel

Reply:
Yes, this was a comment from before I saw that it was just moving existing code around